pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -37,11 +37,17 @@ func (in *NodeClaim) ConvertTo(ctx context.Context, to apis.Convertible) error {
 	v1beta1NC := to.(*v1beta1.NodeClaim)
 	v1beta1NC.ObjectMeta = in.ObjectMeta
 
-	in.Status.convertTo((&v1beta1NC.Status))
-	return in.Spec.convertTo(ctx, &v1beta1NC.Spec, in.Annotations[KubeletCompatibilityAnnotationKey])
+	in.Status.convertTo(&v1beta1NC.Status)
+	if err := in.Spec.convertTo(&v1beta1NC.Spec, in.Annotations[KubeletCompatibilityAnnotationKey], in.Annotations[NodeClassReferenceAnnotationKey]); err != nil {
+		return err
+	}
+	// Remove the annotations from the v1beta1 NodeClaim on the convert back
+	delete(v1beta1NC.Annotations, KubeletCompatibilityAnnotationKey)
```

Comment:
Was this not correct before then?

Reply:
Correct -- we had a brief discussion and realized that keeping the annotations around here is not the correct behavior.

Reply:
The compatibility annotation was there for observability, since this an annotation that is called out in the docs that customer will need drop when going to v1.1 or rolling back.